File: array/missing_ranges/missing_ranges.rb
require 'set'

def missing_ranges(arr, low, high)
  hashed = Set.new arr
  (low...high).each do |n|
    puts n unless hashed.include?(n)
  end
end

inpt = [10, 12, 11, 15]
low = 10
hi = 15
p('input: ', inpt)
p('result: ')
missing_ranges(inpt, low, hi)

File: array/missing_ranges/missing_ranges.py
def missing_ranges(arr, low, high):
    hashed = set(arr)
    for n in range(low, high):
        if n not in hashed:
            print(n)


inpt = [10, 12, 11, 15]
low, hi = 10, 15
print("input: ", inpt)
print("result: ")
missing_ranges(inpt, low, hi)